third_party/move/move-model/src/lib.rs

```diff
@@ -61,10 +61,49 @@ pub mod ty;
 pub mod well_known;
 
 // =================================================================================================
-// Entry Point
+// Entry Point V2
+
+/// Represents information about a package: the sources it contains and the package private
+/// address mapping.
+#[derive(Debug, Clone)]
+pub struct PackageInfo {
+    pub sources: Vec<String>,
+    pub address_map: BTreeMap<String, NumericalAddress>,
+}
 
-/// Build the move model with default compilation flags and default options and no named addresses.
-/// This collects transitive dependencies for move sources from the provided directory list.
+/// Builds the Move model for the v2 compiler. This builds the model, compiling both code
+/// and specs from sources into the typed check AST. No bytecode is attached to the model.
```

Comment:
"typed check AST" doesn't parse for me.  Is there a "check AST" which has been typed?  Or should it be a typechecked AST?